account_invoice_import_facturae/wizard/account_invoice_import.py

```diff
@@ -0,0 +1,144 @@
+# © 2015-2017 Akretion (Alexis de Lattre <alexis.delattre@akretion.com>)
+# License AGPL-3.0 or later (https://www.gnu.org/licenses/agpl).
+
+
+from datetime import datetime
+import logging
+from lxml import etree
+import mimetypes
+
+from odoo import api, fields, models, tools, _
+from odoo.exceptions import ValidationError
+logger = logging.getLogger(__name__)
+
+try:
+    import xmlsig
+except(ImportError, IOError) as err:
+    logging.info(err)
+
+facturae_ns = 'http://www.facturae.es/Facturae/2009/v3.2/Facturae'
+
+
+class AccountInvoiceImport(models.TransientModel):
+    _inherit = 'account.invoice.import'
+
+    @api.multi
+    def parse_invoice(self):
+        mimetypes.add_type('application/xml', '.xsig')
+        return super().parse_invoice()
+
+    @api.model
+    def parse_xml_invoice(self, xml_root):
+        if (
+            xml_root.tag and
+            xml_root.tag.startswith('{%s}Facturae' % facturae_ns)
+        ):
+            return self.parse_facturae_invoice(xml_root)
+        return super().parse_xml_invoice(xml_root)
+
+    @api.model
+    def parse_facturae_invoice(self, xml_root):
+        facturae_schema = etree.XMLSchema(
+            etree.parse(tools.file_open(
+                "Facturaev3_2.xsd",
+                subdir="addons/account_invoice_import_facturae/data"
+            )))
+        facturae_schema.assertValid(xml_root)
+        sign = xml_root.find(
+            'ds:Signature', namespaces={'ds': xmlsig.constants.DSigNs}
+        )
+        if sign is None:
+            raise ValidationError(_('Signature node cannot be found'))
```

Comment:
Would be possible to import unsigned invoices?

Reply:
It could be possible, but the signature is important, because it certifies that the invoice is not modified.
We could do if sign is not None: xmlsig.SignatureContext().verify(sign)
It is ok?

Reply:
Yes, that is ok. I tried and it worked for both, signed and unsigned